src/tools/miri/tests/pass/float.rs

```diff
@@ -198,7 +198,7 @@ fn casts() {
     assert_eq::<i32>(f32::INFINITY as i32, i32::MAX);
     assert_eq::<i32>(f32::NEG_INFINITY as i32, i32::MIN);
     assert_eq::<i32>(f32::NAN as i32, 0);
-    assert_eq::<i32>((-f32::NAN) as i32, 0);
```

Comment:
I don't think we should lint against `(-expr) as i32`. Looking at `-expr as i32` I have no idea whether the `as` or the `-` are executed first, and I might care about that (in particular when casting to an unsigned type).

Reply:
I agreed :)

Reply:
Thanks! The diff in the Miri test can hopefully be removed then. :)